File: samples/client/3_1_0_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/MinLengthValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.exceptions.ValidationException;
import org.checkerframework.checker.nullness.qual.Nullable;

public class MinLengthValidator extends LengthValidator implements KeywordValidator {
    public final int minLength;

    public MinLengthValidator(int minLength) {
        this.minLength = minLength;
    }

    @Override
    public @Nullable PathToSchemasMap validate(
        ValidationData data
    ) {
        if (!(data.arg() instanceof String stringArg)) {
            return null;
        }
        int length = getLength(stringArg);
        if (length < minLength) {
            throw new ValidationException("Value " + stringArg + " is invalid because has < the minLength of " + minLength);
        }
        return null;
    }
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/MinLengthValidator.kt
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.exceptions.ValidationException;
import org.checkerframework.checker.nullness.qual.Nullable;

public class MinLengthValidator extends LengthValidator implements KeywordValidator {
    @Override
    public @Nullable PathToSchemasMap validate(
        ValidationData data
    ) throws ValidationException {
        var minLength = data.schema().minLength;
        if (minLength == null) {
            return null;
        }
        if (!(data.arg() instanceof String stringArg)) {
            return null;
        }
        int length = getLength(stringArg);
        if (length < minLength) {
            throw new ValidationException("Value " + stringArg + " is invalid because has < the minLength of " + minLength);
        }
        return null;
    }
}